plot-builder-portable/src/commonMain/kotlin/jetbrains/datalore/plot/builder/tooltip/CompositeValue.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2020. JetBrains s.r.o.
+ * Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+ */
+
+package jetbrains.datalore.plot.builder.tooltip
+
+import jetbrains.datalore.plot.base.interact.DataContext
+import jetbrains.datalore.plot.base.interact.ValueSource
+import jetbrains.datalore.plot.base.interact.ValueSource.DataPoint
+
+class CompositeValue(
+    private val values: List<ValueSource>,
+    private val label: String,
+    format: String
+) : ValueSource {
+
+    private val myFormatter = if (format.isEmpty()) null else LineFormatter(format)
+
+    override fun setDataPointProvider(dataContext: DataContext) {
+        values.forEach { it.setDataPointProvider(dataContext) }
+    }
+
+    override fun getDataPoint(index: Int): DataPoint? {
+        val dataValues = values.map { dataValue ->
+            dataValue.getDataPoint(index)
```

Comment:
`dataValue.getDataPoint(index) ?: return null`
You will not need check `null in dataValues` and cast `dataValues as List<DataPoints>`